### PR TITLE
Fixup IMC

### LINKED ISSUE #418: imc tests hang on csg_repack
If you run the IMC tutorials they hang in `csg_imcrepack` using https://github.com/votca/votca/pull/191:
```
cmake .. -DENABLE_TESTING=ON -DENABLE_REGRESSION_TESTING=ON -DENABLE_EXPERIMENTAL_TESTS=ON
make
ctest -V -R LJ
...
solving linear equations for imc group 'group_1' (regularization 1000)
```

`csg_imcrepack` was only touched in 16e43ee22a497122854ed7de2a6171528163b23b, if I revert that commit it fails on the unpack after running `csg_imc_solve` with:
```
Running critical command 'csg_imcrepack --in group_1.packed --unpack group_1.packed.sol'
terminate called after throwing an instance of 'std::runtime_error'
  what():  invalid range 61:60: begin, end and stride do not form a closed interval
```

### PR DESCRIPTION
**Pull request by junghans**
Fix #418

### CODE REVIEW

**Comment by junghans** [comment]
```
      Start  95: Compare_csg_repack-unpack_output_1
10/27 Test  #95: Compare_csg_repack-unpack_output_1 .................   Passed    0.07 sec
      Start  96: Compare_csg_repack-unpack_output_2
11/27 Test  #96: Compare_csg_repack-unpack_output_2 .................   Passed    0.07 sec
      Start  97: Compare_csg_repack-unpack_output_3
12/27 Test  #97: Compare_csg_repack-unpack_output_3 .................   Passed    0.16 sec
```
which is good, but
```
	 91 - Compare_csg_repack_output_1 (Failed)
	 92 - Compare_csg_repack_output_2 (Failed)
	 93 - Compare_csg_repack_output_3 (Failed)
```
which is bad.

So packing is brocken, but  unpacking is fine.

**Comment by DRosen285** [comment]
Don't know if this really helps, but when I comment out "imcio_write_matrix(name_out + ".gmc", gmc, &list);"
 packing does not hang anymore.
Nevertheless the packed.imc file that gets created still looks weird to me (I'm running the LJ example). So maybe there is the sth. going wrong with "imcio_write_dS" and "imcio_write_matrix".

**Comment by JensWehner** [comment]
Yeah I probably messed the read and write up.

**Comment by junghans** [comment]
@JensWehner it looks like the pack and unpack was done in order to workaround zeros in matrix, however due to the fact that we had to limit the iteration range to non-zero rdf regions for different reasons, this might not be needed anymore. I just tested this with all tutorials. Thanks to @DRosen285 for the hint and discussion on that.

Could you merge the unpack part (writing the .dpot.imc files) into csg_imc_solve ? And then we just can remove csg_imcrepack fully.

File: src/tools/csg_imc_solve.cc [hunk 1]
```diff
@@ -104,4 +107,19 @@ void CG_IMC_solve::Run() {
   x.y() = -inverse * A.transpose() * B.y();
 
   x.Save(outputfile);
```
Comment (junghans) [inline src/tools/csg_imc_solve.cc]: @DRosen285 is the output of the complete solution vector still useful if we also write the potentials?

File: src/tools/csg_imc_solve.cc [hunk 2]
```diff
@@ -104,4 +107,19 @@ void CG_IMC_solve::Run() {
...
+
+  if (OptionsMap().count("idxfile")) {
+    std::string idxfile = _op_vm["idxfile"].as<std::string>();
+    std::vector<std::pair<std::string, votca::tools::RangeParser> > ranges =
+        votca::csg::imcio_read_index(idxfile);
+    votca::tools::Table tbl_in;
+    tbl_in.Load(outputfile);
```
Comment (junghans) [inline src/tools/csg_imc_solve.cc]: @JensWehner maybe you can think about a better solution here than reading it in again.

File: src/tools/csg_imc_solve.cc (hunk 1 above)
Comment (DRosen285) [reply src/tools/csg_imc_solve.cc]: If with potentials you mean the potential update (dU or dpot) as well, then I think this can go away.

**Comment by JensWehner** [comment]
What does it take to close this PR? I am a bit confused, I do not mind the C++ work, but I have no idea what is supposed to be going on? Can someone tell me, what is supposed to be done.

File: src/tools/csg_imc_solve.cc (hunk 2 above)
Comment (JensWehner) [reply src/tools/csg_imc_solve.cc]: Like this?

File: src/tools/csg_imc_solve.cc (hunk 1 above)
Comment (junghans) [reply src/tools/csg_imc_solve.cc]: Then we should move this, too!

File: src/tools/csg_imc_solve.cc (hunk 2 above)
Comment (junghans) [reply src/tools/csg_imc_solve.cc]: Superb!

File: src/tools/csg_imc_solve.cc (hunk 1 above)
Comment (junghans) [reply src/tools/csg_imc_solve.cc]: Done.

**Comment by junghans** [comment]
@JensWehner I think now this is ready to merge.

**Review by DRosen285 — APPROVED** [review]

**Comment by JensWehner** [comment]
if you guys like it then I am good with it

**Review by JensWehner — APPROVED** [review]

**Pull request merged**